File: day1/p1.py
def rotate(x):
    return int(x.replace("R", "").replace("L", "-"))


with open("assets/submit.txt", "r", encoding="utf-8") as f:
    raw_text = f.read().splitlines()

loc, res = 50, 0

for rotation in raw_text:
    loc = (loc + rotate(rotation)) % 100
    if loc == 0:
        res += 1

print(res)

File: day1/p1.go
package main

import (
	"fmt"
	"os"
	"log"
	"strings"
	"strconv"
)

func main() {
	content, err := os.ReadFile("assets/submit.txt")
	if err != nil {
		log.Fatal(err)
	}

	lines := strings.Split(string(content), "\n")
	replacer := strings.NewReplacer("R", "", "L", "-")

	var pos = 50
	var res int

	for _, line := range lines {
		if line == "" {
			continue
		}
		i, err := strconv.Atoi(replacer.Replace(line))
		if err != nil {
			log.Fatal(err)
		}

		pos += i
		pos %= 100

		if pos == 0 {
			res += 1
		}
	}

	fmt.Println(res)
}
